File: src/Roguelike.Core/Core/Room/RestRoomHandler.cs
using Roguelike.Core.Map;
using System;

namespace Roguelike.Core.Handlers
{
    public class RestRoomHandler : IRoomHandler
    {
        public void Execute(GameRun run, Room room)
        {
            float baseHealPercentage = 0.30f;
            
            int healAmount = (int)Math.Floor(run.TheHero.MaxHealth * baseHealPercentage);
            run.TheHero.Heal(healAmount);
            run.CurrentState = GameState.OnMap;
        }
    }
}

File: python_roguelike/core/rooms/rest_room_handler.py
import math
from .i_room_handler import IRoomHandler
from ...data.enums import GameState


class RestRoomHandler(IRoomHandler):
    def execute(self, run, room):
        heal_pct = getattr(run, 'rest_heal_pct', 0.30)
        heal_amount = int(math.floor(run.the_hero.max_health * heal_pct))
        run.the_hero.heal(heal_amount)
        run.current_state = GameState.OnMap
